### PR TITLE
Dashboard: support parse visual plan

### PR DESCRIPTION
**Pull request by srstack**
support parse visual plan

### CODE REVIEW

**Comment by srstack** [comment]
@shhdgit

File: pkg/apiserver/utils/visual_plan.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	// proto unmarshal
+	visual := &tipb.VisualData{}
+	err = visual.Unmarshal(vpBytes)
+	if err != nil {
+		return nil, err
+	}
```
Comment (shhdgit) [inline pkg/apiserver/utils/visual_plan.go:37]: It's wired that the impl of the compression algorithm and encoding algo are leaked from tidb.

Comment (shhdgit) [reply pkg/apiserver/utils/visual_plan.go:37]: Why does tidb not provide the `tidb_decode_visual_plan` function as the design doc says?

**Review by shhdgit — APPROVED** [review]

File: pkg/apiserver/slowquery/service.go [hunk 2]
```diff
@@ -99,6 +99,14 @@ func (s *Service) getDetails(c *gin.Context) {
 		rest.Error(c, err)
 		return
 	}
+
+	// get visual plan
+	result.VisualPlan, err = utils.GenerateVisualPlanJSON(result.VisualPlan)
+	if err != nil {
+		rest.Error(c, err)
+		return
+	}
```
Comment (breezewish) [inline pkg/apiserver/slowquery/service.go:108]: Will this work with v6.1?

Comment (shhdgit) [reply pkg/apiserver/slowquery/service.go:108]: No, it's merged into the feature branch.

Comment (breezewish) [reply pkg/apiserver/slowquery/service.go:108]: Does this mean that we need to cherry pick instead of fast-forward when publishing new releases?

Comment (breezewish) [reply pkg/apiserver/slowquery/service.go:108]: It will be horrible if the master branch of TiDB Dashboard does not work with the master branch of TiDB.

Comment (breezewish) [reply pkg/apiserver/slowquery/service.go:108]: Oh, I got you. This PR is not merging into master.

Comment (shhdgit) [reply pkg/apiserver/slowquery/service.go:108]: Though the `FeatVisualPlan` branch is designed for collaboration of frontend and backend and better review experiences.

Maybe cherry pick for the specific release branches will be better in the future.

Comment (shhdgit) [reply pkg/apiserver/slowquery/service.go:108]: > Oh, I got you. This PR is not merging into master.

yes~

**Pull request merged**